### PR TITLE
[Merged by Bors] - feat: port Order.OrderIsoNat

### PR DESCRIPTION
**Pull request by zaxioms0**
---

Just the beginning

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/OrderIsoNat.lean [hunk 1]
```diff
@@ -112,24 +113,26 @@ variable (s : Set ℕ) [Infinite s]
...
+  classical
+  exact
       RelIso.ofSurjective
         (RelEmbedding.orderEmbeddingOfLTEmbedding
           (RelEmbedding.natLt (Nat.Subtype.ofNat s) fun n => Nat.Subtype.lt_succ_self _))
         Nat.Subtype.ofNat_surjective
 #align nat.subtype.order_iso_of_nat Nat.Subtype.orderIsoOfNat
 
-variable {s}
+--porting note: Added the decidability requirement, I'm not sure how it worked in lean3 without it
```
Comment (jcommelin) [inline Mathlib/Order/OrderIsoNat.lean:131]: Do you need it for the statements, or for the proofs?

Comment (zaxioms) [reply Mathlib/Order/OrderIsoNat.lean:131]: For the statements. In particular, `orderEmbeddingOfSet` requires it

**Comment by zaxioms** [comment]
(I think Im supposed to add back the awaiting review tag? sorry if not)

File: Mathlib/Order/OrderIsoNat.lean (hunk 1 above)
Comment (ChrisHughes24) [reply Mathlib/Order/OrderIsoNat.lean:131]: This is the `classical` leakage in Lean3 again

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**